### PR TITLE
Extract log collector crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
The program-runtime crate is too big

#### Summary of Changes
Extract log_collector.rs into its own crate

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,7 @@ target/
 *.log
 log-*.txt
 log-*/
+!log-collector/
```
Comment (buffalojoec) [inline .gitignore:20]: What's this needed for?

**Review by buffalojoec — APPROVED** [review]
Lgtm, thanks!

File: .gitignore (hunk 1 above)
Comment (kevinheavey) [reply .gitignore:20]: Without this the directory for the new crate gets ignored

Comment (buffalojoec) [reply .gitignore:20]: Ohh the line above, duh. Thanks.

**Comment by yihau** [comment]
thank you!

**Pull request merged**